cmd/gaia/app/genesis.go

```diff
@@ -171,27 +83,33 @@ func GaiaAppGenState(cdc *codec.Codec, appGenTxs []json.RawMessage) (genesisStat
 
 	// start with the default staking genesis state
 	stakeData := stake.DefaultGenesisState()
-
 	slashingData := slashing.DefaultGenesisState()
 
 	// get genesis flag account information
 	genaccs := make([]GenesisAccount, len(appGenTxs))
-	for i, appGenTx := range appGenTxs {
 
-		var genTx GaiaGenTx
-		err = cdc.UnmarshalJSON(appGenTx, &genTx)
+	for i, genTx := range appGenTxs {
+		var tx auth.StdTx
+		err = cdc.UnmarshalJSON(genTx, &tx)
 		if err != nil {
 			return
 		}
+		msgs := tx.GetMsgs()
+		if len(msgs) == 0 {
+			err = errors.New("must provide at least genesis message")
+			return
+		}
+		msg := msgs[0].(stake.MsgCreateValidator)
 
 		// create the genesis account, give'm few steaks and a buncha token with there name
-		genaccs[i] = genesisAccountFromGenTx(genTx)
+		genaccs[i] = genesisAccountFromMsgCreateValidator(msg, freeFermionsAcc)
```

Comment:
I guess this part is pretty Gaia-specific (as is needing to update the stake data).

Reply:
But also, this won't *actually* be in Gaia, since we'll have an account set unrelated to the validator transactions. This needs to move to a special `testnet` command I think.

Reply:
I think the approach that Gaia will use (separate account set) is likely to be common - and we might want to allow delegation transactions at genesis too.

Reply:
Maybe this can be split into a separate PR. Also cc @zmanian - what do we need, if anything, for GoS genesis setup?

Reply:
I guess we can use this for now and just strip the accounts generated here and use the fundraiser distribution instead, copy/paste job.